src/main/groovy/nine/jasper/spring/AbstractJasperReportsView.groovy

```diff
@@ -263,14 +266,37 @@ public abstract class AbstractJasperReportsView extends AbstractUrlBasedView {
 	 * report has been statically defined
 	 */
 	protected JasperReport loadReport() {
-		String url = getUrl();
+		Resource resource = getResource()
+		if(resource == null) return null
+		return JasperUtils.loadReport(resource)
+	}
+
+	/**
+	 * Returns the report as spring resource
+	 * @return Resource
+	 */
+	Resource getResource() {
+		String url = getUrl()
 		if (url == null) {
-			return null;
+			return null
 		}
 		//FIXME replace this with the viewResourceLoader
 		//it should have the full aboslute URL at this point in the game
-		Resource mainReport = getApplicationContext().getResource(url);
-		return JasperUtils.loadReport(mainReport)
+		return getApplicationContext().getResource(url);
+	}
+
+	/**
+	 * Returns the file object for this report, if exists, or returns null.
+	 * @return
+	 */
+	File getFile() {
+		try {
+			Resource resource = getResource()
+			return resource?.file
```

Comment:
I'm skeptical that this works when running as a WAR file. Calling `file` on a resource loaded from a war file usually throws an exception since there's no physical file object.

Has this been tested in both exploded as well as war?